vars/githubWorkflowRun.groovy

```diff
@@ -0,0 +1,88 @@
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+
+  This step triggers workflow run on Github actions.
+
+  Example:
+     script {
+       def args = [
+           repo: "owner/repository",
+           workflow: "build.yml", 
+           ref: "main",
+           parameters: [
+             path: "filebeat"
+           ],
+           credentialsId: "github-workflow-token",
+           ghVersion: "2.1.0"]
+       def runId = githubWorkflowRun.triggerGithubActionsWorkflow(args)
+       def runInfo = githubWorkflowRun.getWorkflowRun(args + [runId: runId]) 
+   }
+*/
+
+import groovy.json.JsonSlurperClassic 
+
+def call(Map args = [:]) {
+    def buildTimeLimit = args.get('buildTimeLimit', 1)
+    def runId = triggerGithubActionsWorkflow(args)
+    def startDate = new Date()
+    while(startDate.time  + buildTimeLimit*3600000 -  new Date().time > 0) {
+        def runInfo = getWorkflowRun(args + [runId: runId])
+        if (runInfo.status == "completed") return runInfo
+        sleep(300); // sleep 5 min
+    }
+    error("Build time out")
+}
+
+def triggerGithubActionsWorkflow(Map args = [:]) {
+    if (!args.workflow) error('gh: workflow parameter is required.')
+    def ghVersion = args.get("ghVersion", "2.1.0")
+    def ref = args.get("ref", "master")
+    def runner = args.get("runner", "ubuntu-latest")
+    def runId = "${ref}-${new Date().getTime()}-${env.BUILD_ID}"
+    def parameters = args.get("parameters", [:])
+    def inputs = (parameters + [id: runId, runner: runner]).collect{ "${it}" }
+    gh(command: "workflow run ${args.workflow}", credentialsId: args.credentialsId,
+        version: ghVersion, forceInstallation: true,
+        flags: [repo: args.repo, ref: ref, field: inputs ])
+    sleep(30)
+    def runsText = gh(command: "run list", credentialsId: args.credentialsId,
+        version: ghVersion, forceInstallation: false,
+        flags: [repo: args.repo, workflow: args.workflow, limit: 100])
+    def runIds = []
+    for(def s: runsText.split("\n")) {
+        def r = s.split(/\s+/)
+        if (r[-1] == "0m"|| r[-1] == "1m") runIds.add(r[-3]) 
+    }
+    for(def run: runIds) {
+        def runText=gh(command: "run view -v ${run}", credentialsId: args.credentialsId,
+            version: ghVersion, forceInstallation: false, flags: [repo: args.repo])
+        if (runText.split("\n").find { it ==~ /^\s+\S\s+Run ID ${runId}\s*$/} ) {
+            return run as int
+        }
+    }
+    return 0
+}
+
+def getWorkflowRun(Map args = [:]) {
```

Comment:
If this method can be called independently, then `version: ghVersion, forceInstallation: false` might be requited to be changed to `version: ghVersion, forceInstallation: true`.

At the moment, the `gh` installation does not verify if the given `gh` version matches with the existing installed one. 

So, I'd say it might be good to install regardless to avoid any side effects.

Reply:
What I commented above won't be needed since https://github.com/elastic/apm-pipeline-library/pull/1364 will help to run version validation within the `gh` step.

Reply:
Fixed